File: api/src/lib/mappers/whatsapp-log.mapper.ts
import type { WhatsappLog } from "../generated/prisma/client.js";
import type { IWhatsappLog } from "../types/domain/index.js";

export function toWhatsappLogEntity(p: WhatsappLog): IWhatsappLog {
  return {
    id: p.id,
    received_at: p.receivedAt,
    device: p.device,
    mode: p.mode,
    sender: p.sender,
    sender_lid: p.senderLid,
    sender_name: p.senderName,
    is_group: p.isGroup,
    group_id: p.groupId,
    member_phone: p.memberPhone,
    member_lid: p.memberLid,
    message_text: p.messageText,
    message_type: p.messageType,
    is_forwarded: p.isForwarded,
    is_quick: p.isQuick,
    inbox_id: p.inboxId,
    extension: p.extension,
    filename: p.filename,
    url: p.url,
    location: p.location,
    poll_name: p.pollName,
    poll_choices: p.pollChoices,
    wa_timestamp: p.waTimestamp.toString(),
  };
}

File: handlerclaw/lib/features/whatsapp-logs/data/mappers/whatsapp_log_mapper.dart
import 'package:handlerclaw/features/whatsapp-logs/data/dto/whatsapp_log_dto.dart';
import 'package:handlerclaw/features/whatsapp-logs/domain/entities/whatsapp_log_entity.dart';

class WhatsappLogMapper {
  static WhatsappLogEntity fromDto(WhatsappLogDto dto) {
    return WhatsappLogEntity(
      id: dto.id,
      receivedAt: dto.receivedAt,
      device: dto.device,
      mode: dto.mode,
      sender: dto.sender,
      senderLid: dto.senderLid,
      senderName: dto.senderName,
      isGroup: dto.isGroup,
      groupId: dto.groupId,
      memberPhone: dto.memberPhone,
      memberLid: dto.memberLid,
      messageText: dto.messageText,
      messageType: dto.messageType,
      isForwarded: dto.isForwarded,
      isQuick: dto.isQuick,
      inboxId: dto.inboxId,
      extension: dto.extension,
      filename: dto.filename,
      url: dto.url,
      location: dto.location,
      pollName: dto.pollName,
      pollChoices: dto.pollChoices,
      waTimestamp: dto.waTimestamp,
    );
  }

  static List<WhatsappLogEntity> fromDtos(List<WhatsappLogDto> dtos) {
    return dtos.map((dto) => fromDto(dto)).toList();
  }
}
